### ISSUE TITLE
Create BindingResult objects for iterated objects

### ISSUE THREAD

**Issue by danielfernandez** [thymeleaf/thymeleaf-spring#49]
In order to try to support scenarios like the ones described at #37 and #44, we could study the possibility of automatically generating corresponding `BindingResult` objects for the iterated objects in `th:each` processors.

The code for adding these `BindingResult` objects should probably be very similar to what is currently being done at the Thymeleaf Testing framework here https://github.com/thymeleaf/thymeleaf-testing/blob/thymeleaf-testing-2.1.2.RELEASE/src/main/java/org/thymeleaf/testing/templateengine/context/web/SpringWebProcessingContextBuilder.java#L336 

Different aspects like performance impact, possible name collisions, low-level binding to non-public Spring code, etc. should be studied before adding this code. Also, tests should be done to determine how these _artificially created_ objects could affect normal form attribute processing (we would be creating a `BindingResult` for an object contained in another object that already has a `BindingResult`).

**Comment by cocevk** [comment]
Hi, Daniel. This "bug" has wasted a lot of my time and probably will waste a lot of time for others, too. Please fix asap. Otherwise, thymeleaf is the ULTIMATE framework! :)

**Comment by danielfernandez** [comment]
This has been solved by taking a different, more Spring-integrated approach.

Modifications have been done on Thymeleaf's `SpelVariableExpressionEvaluator` class so that, if a type conversion is needed, instead of executing the expression and then applying the conversion service, Spring's SpEL engine is already called asking for a result of type `String.class`.

This has the effect of allowing the Spring EL engine to examine the expression level by level, checking for any conversion metadata needed to be applied at any of these levels (like e.g. `@DateTimeFormat` or `@NumberFormat` annotations at the class of an iterated object). The Spring EL engine will in this case correctly create `TypeDescriptor` objects for the execution targets containing these conversion metadata, and therefore the result will be as expected.

In summary: more work for Spring, less for Thymeleaf, better results ;-)

A test has been produced for this scenario: https://github.com/thymeleaf/thymeleaf-tests/blob/eecdb14dafc5a4e06a29c8a2ff8468d0b2fab1ff/src/test-21/resources/engine21/conversion/conversion3/conversion3-06.thtest

**Comment by danielfernandez** [comment]
This feature will be a part of 2.1.3.

2.1.3-SNAPSHOT updated.
